decidim-budgets/app/controllers/decidim/budgets/admin/application_controller.rb

```diff
@@ -16,15 +16,15 @@ def budget
         end
 
         def projects
-          return unless budget
-
-          @projects ||= budget.projects
+          @projects ||= if budget
+                          budget.projects
+                        else
+                          Decidim::Budgets::Project.joins(:budget).where(budget: { component: current_component })
+                        end
         end
 
         def project
-          return unless projects
-
-          @project ||= projects.find(params[:id])
+          @project ||= projects.find(params[:project_id] || params[:id])
```

Comment:
The `projects` is already guaranteed to be in the correct scope if we are fetching the project by ID from the budget's list, it should be already fine as is.